### PR TITLE
Added electric field sensor driver

### PR DESCRIPTION
**Pull request by eaeyal**

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -63,6 +63,13 @@
       "driver_class": "LEDDriver",
       "device_id": "MISC_2",
       "enabled": true
+    },
+
+    {
+      "name": "electric_field_sensor",
+      "driver_class": "ElectricFieldSensor",
+      "device_id": "MISC_3",
```
Comment (DarylDohner) [inline config.json:71]: should be MISC_SENSOR_3

File: config.json [hunk 2]
```diff
@@ -63,6 +63,13 @@
       "driver_class": "LEDDriver",
       "device_id": "MISC_2",
       "enabled": true
+    },
+
+    {
+      "name": "electric_field_sensor",
```
Comment (DarylDohner) [inline config.json:69]: use hyphens not underscores

File: EosPayload/drivers/efield_driver.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        ADC.setup(self.pin_2)
+        ADC.setup(self.pin_3)
+
+    def device_read(self, logger: logging.Logger) -> None:
+        adc_pins = [self.pin_1, self.pin_2, self.pin_3]
+        voltages = [None, None, None]
+        while True:
+            try:
+                # Read the voltage from the ADC pin
+                voltages = [ADC.read(pin) for pin in adc_pins]
```
Comment (DarylDohner) [inline EosPayload/drivers/efield_driver.py:47]: you should restructure this into a loop so you can read the pins independently.  that way if an exception is raised on reading pin1, we can still try to read pins 2 and 3

File: EosPayload/drivers/efield_driver.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            header = DataHeader(
+                data_type=Type.E_FIELD,
+                sender=self.get_device_id(),
+                priority=Priority.DATA
+            )
+            packet = Packet(
+                body=efield_obj,
+                data_header=header,
+            )
+            self._mqtt.send(Topic.RADIO_TRANSMIT, packet)
```
Comment (DarylDohner) [inline EosPayload/drivers/efield_driver.py:77]: include this stuff in your try / catch as well

File: EosPayload/drivers/efield_driver.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+            self.thread_sleep(logger, 2)
```
Comment (DarylDohner) [inline EosPayload/drivers/efield_driver.py:79]: do you want to read data and transmit data at the same rate?  How rapidly are the values expected to change for this sensor?  If it changes more rapidly, you may want to read and data_log every 1s or 0.5s.  How valuable is a real-time measurement at the ground station?  If its something that could suddenly spike, maybe we want to transmit every 1s.  if it's something that is going to be relatively stable, then maybe we dont need so many points on the live graph and you transmit every 5s.  Maybe log+transmit every 2s is what works for this.  Just something for you to think about

**Review by DarylDohner — CHANGES_REQUESTED** [review]
see comments

File: EosPayload/drivers/efield_driver.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+                        data_type=Type.E_FIELD,
+                        sender=self.get_device_id(),
+                        priority=Priority.DATA
+                    )
+                    packet = Packet(
+                        body=efield_obj,
+                        data_header=header,
+                    )
+                    self._mqtt.send(Topic.RADIO_TRANSMIT, packet)
+                    self.thread_sleep(logger, 1)
```
Comment (DarylDohner) [inline EosPayload/drivers/efield_driver.py:78]: this thread sleep needs to happen outside this if block, otherwise you are only sleeping on transmit

**Review by DarylDohner — DISMISSED** [review]

**Review by David-Rey — APPROVED** [review]
lgtm

**Pull request merged**